proofs/tree.go

```diff
@@ -280,6 +280,19 @@ func NewDocumentTree(proofOpts TreeOptions) DocumentTree {
 	}
 }
 
+// NewDocumentTree returns a DocumentTree with `rootHash`
+func NewDocumentTreeWithRootHash(rootHash []byte, hash hash.Hash) DocumentTree {
```

Comment:
We'll need to support other options (such as the prefix) as well. Best would be to support the same treeoptions as https://godoc.org/github.com/centrifuge/precise-proofs/proofs#NewDocumentTree